modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/SourceHandler.java

```diff
@@ -605,7 +609,8 @@ public void timeout(NHttpServerConnection conn) {
                             + "CAUSE_OF_ERROR = Connection between the client and the EI timeouts, HTTP_URL = "
                             + logDetails.get("url") + ", " + "HTTP_METHOD = " + logDetails.get("method")
                             + ", SOCKET_TIMEOUT = " + conn.getSocketTimeout() + ", CLIENT_ADDRESS = "
-                            + getClientConnectionInfo(conn) + ", CONNECTION " + conn);
+                            + getClientConnectionInfo(conn) + ", CONNECTION " + conn + " Correlation ID : "
+                            + conn.getContext().getAttribute(CorrelationConstants.CORRELATION_ID).toString());
```

Comment:
this leads to following NPE in MI

```bash
[2021-04-13 20:11:39,481] ERROR {SourceHandler} - Unexpected error: null java.lang.NullPointerException
	at org.apache.synapse.transport.passthru.SourceHandler.timeout(SourceHandler.java:614)
	at org.apache.synapse.transport.passthru.core.MultiListenerServerIODispatch.onTimeout(MultiListenerServerIODispatch.java:105)
	at org.apache.synapse.transport.passthru.core.MultiListenerServerIODispatch.onTimeout(MultiListenerServerIODispatch.java:32)
	at org.apache.http.impl.nio.reactor.AbstractIODispatch.timeout(AbstractIODispatch.java:175)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.sessionTimedOut(BaseIOReactor.java:261)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.timeoutCheck(AbstractIOReactor.java:502)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.validate(BaseIOReactor.java:211)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:280)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591)
	at java.lang.Thread.run(Thread.java:748)
```
@shilmyhasan  did we consider the case where this can be null ?

Reply:
seems this is when invoked via inbound endpoint ( internal)